### ISSUE TITLE
Missing Unit Tests

### ISSUE THREAD

**Issue by stiefeljackal** [GuVAnj8Gv3RJ/NeosAccountDownloader#24]
As this project grows in size, I am concerned that only manual testing is being performed. This is not good as not only are tests not being documented, but also we are just purely manual testing this product, which is very slow and holds up development time. We want to make sure that we are following the testing pyramid and not the testing ice-cream cone.

![image](https://github.com/GuVAnj8Gv3RJ/NeosAccountDownloader/assets/20023996/da68b3ac-32cd-499f-8d4f-eca12411b118)

As mentioned in #17 , you prefer MSTest, which is fine. You should also look into mocking as well as you want to test small, individual components without interference from other components. Not only that, but you should be mocking things such as the file system or the database as unit tests should be fast and independent from these.

One popular mocking framework is Moq, which has been around for a very long time, so you should not be afraid to use it. I can understand your initial concern especially since this might be your first time performing unit tests due to the fact that you want to keep it to MSTest.

**Comment by GuVAnj8Gv3RJ** [comment]
While Unit tests are important to a large project.

Getting Neos assets off of... Neos as fast as possible is a more urgent item. So basically any feature will be prioritized over this.

**Comment by stiefeljackal** [comment]
Of course. I agree that the urgency is there, and if you need to get the product out of the door before the building collapse, then you must make sacrifices. Still, create tests when you can if there is something tricky that needs testing or create tests as an option to avoid hitting the Neos Cloud.
